src/js/player.js

```diff
@@ -4290,6 +4297,88 @@ class Player extends Component {
     return !!this.isAudio_;
   }
 
+  /**
+   * Get the current audioOnlyMode state or set audioOnlyMode to true or false.
+   *
+   * Setting this to `true` will hide all player components except the control bar,
+   * as well as control bar components needed only for video.
+   *
+   * @param {boolean} [value]
+   *         The value to set audioOnlyMode to.
+   *
+   * @return {boolean}
+   *         True if audioOnlyMode is on, false otherwise.
+   */
+  audioOnlyMode(value) {
+    if (value === undefined) {
+      return this.audioOnlyMode_;
+    }
+
+    if (value === this.audioOnlyMode_) {
+      return;
+    }
+
+    if (typeof value !== 'boolean') {
+      log.warn('audioOnlyMode can only be enabled or disabled using a boolean value');
+      return;
+    }
+
+    if (value === true) {
+      // Update styling immediately so we can get the control bar's height
+      this.addClass('vjs-audio-only-mode');
+
+      const playerChildren = this.children();
+      const controlBar = this.getChild('ControlBar');
+      const controlBarHeight = controlBar && controlBar.height();
+
+      // Hide all player components except the control bar. Control bar components
+      // needed only for video are hidden with CSS
+      playerChildren.forEach(child => {
+        if (child.name_ === 'ControlBar') {
+          return;
+        }
+
+        if (child.el_ && !child.hasClass('vjs-hidden')) {
+          child.hide();
+
+          this.hiddenPlayerChildren_.push(child);
+        }
+      });
+
+      // Set the player height the same as the control bar
+      this.height(controlBarHeight);
+
+      // Show control bar
+      this.userActive(true);
+
+      // Fullscreen is not supported in audioOnlyMode, so exit if we need to
+      if (this.isFullscreen()) {
+        this.exitFullscreen();
+      }
+
+      this.audioOnlyMode_ = true;
+
+      this.trigger('audioonlymodechange');
+    } else {
+      this.removeClass('vjs-audio-only-mode');
+
+      // Show player components that were previously hidden
+      if (this.hiddenPlayerChildren_.length > 0) {
```

Comment:
I think we can safely assume this is an array, so not bother testing `length`. An empty array will simply have no effect.